### DISCUSSION TITLE
How to connect with mac address instead off ssid name

### DISCUSSION THREAD

**Discussion by zectlist** [pschatzmann/ESP32-A2DP#598] [Q&A] [answered]
I want to connect bluetooth audio from esp32 to tws device. but in the my factory use many same type of tws. can u help me? thanks
i use this code
#include "BluetoothA2DPSource.h"
#include "StarWars30.h"
extern const uint8_t StarWars10_raw[];
extern const unsigned int StarWars10_raw_len;

BluetoothA2DPSource a2dp_source;
SoundData *music = new OneChannelSoundData((int16_t*)StarWars30_raw, StarWars30_raw_len/2);

void setup() {
  a2dp_source.start("HY-T26");  
  a2dp_source.write_data(music);
}

void loop() {
}

**Comment by pschatzmann** [comment, marked as answer]
At any point of time you can only be connected to one data sink only!

If you want to connect to potentially different sinks, just look at the start method in the [class documentation](https://pschatzmann.github.io/ESP32-A2DP/html/class_bluetooth_a2_d_p_source.html): you can provide a vector! The system then just connects to the first device in the vector that is available...

You can achieve a similar result by defining an [ssid_callback](https://github.com/pschatzmann/ESP32-A2DP/blob/main/examples/bt_music_sender_ssid_callback/bt_music_sender_ssid_callback.ino): there you just define a callback method where you define if you want to accept a connection based on the provided ssid and rssi. If the callback just returns true it connects to any device it finds first.
